3-UsingAPIs/Week1/homework/ex1-johnWho.js

```diff
@@ -8,21 +8,20 @@ Rewrite this function, but replace the callback syntax with the Promise syntax:
 - If the Promise `rejects`, pass an error as the argument to reject with: "You 
   didn't pass in a first name!"
 ------------------------------------------------------------------------------*/
-// TODO see above
-const getAnonName = (firstName, callback) => {
-  setTimeout(() => {
-    if (!firstName) {
-      callback(new Error("You didn't pass in a first name!"));
-      return;
-    }
-
-    const fullName = `${firstName} Doe`;
-
-    callback(fullName);
-  }, 1000);
+const getAnonName = (firstName) => {
+  const fullName = `${firstName} Doe`;
+  return new Promise((resolve, reject) => {
+    setTimeout(() => {
+      if (firstName) {
+        resolve(fullName);
+      } else {
```

Comment:
Minor improvement: as you are checking that `firstName` is defined in line 15 it would make sense to postpone constructing the full name until after you have established that `firstName` is defined:

```suggestion
  return new Promise((resolve, reject) => {
    setTimeout(() => {
      if (firstName) {
        const fullName = `${firstName} Doe`;
        resolve(fullName);
      } else {
```

Reply:
Agree, this makes more sense